backend/test/services/release.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { describe, expect, test, vi } from 'vitest'
+
+import { createRelease, deleteRelease, getModelReleases, getReleaseBySemver } from '../../src/services/v2/release.js'
+
+const arrayAsyncFilter = vi.hoisted(() => ({
+  asyncFilter: vi.fn(() => []),
+}))
+vi.mock('../../src/utils/v2/array.js', () => arrayAsyncFilter)
+
+const authorisationMocks = vi.hoisted(() => ({
+  userReleaseAction: vi.fn(() => true),
+}))
+vi.mock('../../src/connectors/v2/authorisation/index.js', async () => ({
+  ...((await vi.importActual('../../src/connectors/v2/authorisation/index.js')) as object),
+  default: authorisationMocks,
+}))
+
+const modelMocks = vi.hoisted(() => ({
+  getModelById: vi.fn(),
+}))
+vi.mock('../../src/services/v2/model.js', () => modelMocks)
+
+const releaseModelMocks = vi.hoisted(() => {
+  const obj: any = {}
+
+  obj.aggregate = vi.fn(() => obj)
+  obj.match = vi.fn(() => obj)
+  obj.sort = vi.fn(() => obj)
+  obj.lookup = vi.fn(() => obj)
+  obj.append = vi.fn(() => obj)
+  obj.find = vi.fn(() => obj)
+  obj.findOne = vi.fn(() => obj)
+  obj.updateOne = vi.fn(() => obj)
+  obj.save = vi.fn(() => obj)
+  obj.delete = vi.fn(() => obj)
+
+  const model: any = vi.fn(() => obj)
+  Object.assign(model, obj)
+
+  return model
+})
+vi.mock('../../src/models/v2/Release.js', () => ({ default: releaseModelMocks }))
+
+describe('services > release', () => {
+  test('createRelease > simple', async () => {
+    modelMocks.getModelById.mockResolvedValue(undefined)
+
+    await createRelease({} as any, {} as any)
+
+    expect(releaseModelMocks.save).toBeCalled()
+    expect(releaseModelMocks).toBeCalled()
+  })
+
+  test('createRelease > bad authorisation', async () => {
+    authorisationMocks.userReleaseAction.mockResolvedValueOnce(false)
+    expect(() => createRelease({} as any, {} as any)).rejects.toThrowError(/^You do not have permission/)
```

Comment:
For cases like this we could also assert that release.save() does not get called as well as the error getting thrown as those both seem like the two most important outcomes of someone not having the permission to do something like this?